lgc/state/PalMetadata.cpp

```diff
@@ -527,11 +523,25 @@ void PalMetadata::finalizePipeline() {
     }
   }
 
-  // If there are root user data nodes but none of them are used, adjust userDataLimit accordingly.
-  if (m_userDataLimit->getUInt() == 0 && !m_pipelineState->getUserDataNodes().empty())
+  if (pipelineRequiresAllUserData())
     setUserDataLimit();
 }
 
+// =====================================================================================================================
+// Returns true of the setting in the PAL meta data require all of the user data nodes.
+//
+bool PalMetadata::pipelineRequiresAllUserData() const {
+  // If there are root user data nodes but none of them are used, then PAL wants a non-zoro user data limit.
+  if (m_userDataLimit->getUInt() == 0 && !m_pipelineState->getUserDataNodes().empty())
+    return true;
+
+  // If the spill is used, the entire user data table is considered used.
+  if (userDataNodesAreSpilled())
+    return true;
+
+  return false;
```

Comment:
tiny nit: this could be just `return userDataNodesAreSpilled()`

Reply:
I can change it if other prefer it, but I prefer it this way because it make debugging easier.  I set a break point on the return statement I want very easily.